### PR TITLE
Add "part" directive

### LINKED ISSUE #271: How to build 'part (of)' expression?
Code like `part (of) 'xxx.dart';`, what should i do?

### PR DESCRIPTION
**Pull request by smkhalsa**
fixes #271

### CODE REVIEW

File: test/directive_test.dart [hunk 1]
```diff
@@ -39,6 +40,7 @@ void main() {
       equalsDart(r'''
```
Comment (smkhalsa) [inline test/directive_test.dart:40]: This test is failing because of issues with whitespace.

Comment (smkhalsa) [reply test/directive_test.dart:40]: @natebosch any ideas why this may be happening?

Comment (smkhalsa) [reply test/directive_test.dart:40]: I've skipped this test for now

**Comment by smkhalsa** [comment]
I've rebased on https://github.com/dart-lang/code_builder/pull/269, updated the changelog, and bumped the version.

Can we get this merged, please?

**Comment by natebosch** [comment]
cc @matanlurey - I'm curious for your thoughts on this.

I think we might need to add sorting at some point here. I'm pretty sure a `part` directive can't be followed by an `import` or `export`...

**Comment by smkhalsa** [comment]
closing in favor of #269

**Comment by smkhalsa** [comment]
@natebosch sorting works out of the box with the existing [compare override](https://github.com/smkhalsa/code_builder/blob/part-directive/lib/src/specs/directive.dart#L121) since part is listed last in the `DirectiveType` enum. However, it does require that the `orderDirectives` parameter of `DartEmitter` be set to true.

**Pull request closed without merging**